cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -592,45 +480,52 @@ func (m *McmManager) updateAnnotationOnMachine(ctx context.Context, mcName strin
 		return true, err
 	}
 	clone := machine.DeepCopy()
-	if clone.Annotations != nil {
-		if clone.Annotations[key] == val {
-			klog.Infof("Machine %q priority is already set to 1, hence skipping the update", machine.Name)
-			return false, nil
-		}
-		clone.Annotations[key] = val
-	} else {
+	if clone.Annotations == nil {
 		clone.Annotations = make(map[string]string)
-		clone.Annotations[key] = val
 	}
+	clone.Annotations[key] = val
 	_, err = m.machineClient.Machines(machine.Namespace).Update(ctx, clone, metav1.UpdateOptions{})
 	if err == nil {
 		klog.Infof("Machine %s marked with priority %s successfully", mcName, val)
 	}
 	return true, err
 }
 
-// scaleDownMachineDeployment scales down the machine deployment by the provided scaleDownAmount and returns the updated spec.Replicas after scale down.
-func (m *McmManager) scaleDownMachineDeployment(ctx context.Context, mdName string, scaleDownAmount int) (bool, error) {
-	md, err := m.machineDeploymentLister.MachineDeployments(m.namespace).Get(mdName)
+// scaleDownMachineDeployment scales down the MachineDeployment for given name by the length of toDeleteMachineNames
+// It also updates the machines-marked-by-ca-for-deletion annotation on the machine deployment with the list of toDeleteMachineNames
+// NOTE: Callers MUST take the NodeGroup scalingMutex before invoking this method.
+func (m *McmManager) scaleDownMachineDeployment(ctx context.Context, mdName string, toBeDeletedMachineNames []string) (bool, error) {
+	md, err := m.GetMachineDeploymentObject(mdName)
 	if err != nil {
-		klog.Errorf("Unable to fetch MachineDeployment object %s, Error: %v", mdName, err)
 		return true, err
 	}
-	mdclone := md.DeepCopy()
-	expectedReplicas := mdclone.Spec.Replicas - int32(scaleDownAmount)
-	if expectedReplicas == mdclone.Spec.Replicas {
-		klog.Infof("MachineDeployment %q is already set to %d, skipping the update", mdclone.Name, expectedReplicas)
+
+	scaleDownAmount := len(toBeDeletedMachineNames)
+	expectedReplicas := md.Spec.Replicas - int32(scaleDownAmount)
+	if expectedReplicas == md.Spec.Replicas {
```

Comment:
instead this could have been simplified by simply checking the following:
```go
if len(toBeDeleteMachineNames) > 0 {
  // do the processing
} else {
   // log
}
```